clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -3171,18 +3191,38 @@ static void handleSubGroupSize(Sema &S, Decl *D, const ParsedAttr &AL) {
 }
 
 // Handles num_simd_work_items.
-static void handleNumSimdWorkItemsAttr(Sema &S, Decl *D, const ParsedAttr &A) {
+static void handleNumSimdWorkItemsAttr(Sema &S, Decl *D, const ParsedAttr &AL) {
   if (D->isInvalidDecl())
     return;
 
-  Expr *E = A.getArgAsExpr(0);
+  Expr *E = AL.getArgAsExpr(0);
 
   if (D->getAttr<SYCLIntelNumSimdWorkItemsAttr>())
-    S.Diag(A.getLoc(), diag::warn_duplicate_attribute) << A;
+    S.Diag(AL.getLoc(), diag::warn_duplicate_attribute) << AL;
 
-  S.CheckDeprecatedSYCLAttributeSpelling(A);
+  S.CheckDeprecatedSYCLAttributeSpelling(AL);
+
+  if (const auto *A = D->getAttr<ReqdWorkGroupSizeAttr>()) {
+    ASTContext &Ctx = S.getASTContext();
+
+    if (!E->isValueDependent()) {
+      Optional<llvm::APSInt> NumSimdWorkItems = E->getIntegerConstantExpr(Ctx);
+      Optional<llvm::APSInt> XDimVal = A->getXDimVal(Ctx);
```

Comment:
Also, what if any of [XYZ]DimVal is value-dependent? (This may also be an existing bug.)

Reply:
> Also, what if any of [XYZ]DimVal is value-dependent? (This may also be an existing bug.)

Causes assertion, so add assert here.

Reply:
I think this will assert on possibly valid code and we're missing test coverage for it. I think the test cases I added above would trigger the assertions. So I don't think assertions are valid here -- I think we need to change the additional members we added in Attr.td to return an empty optional object if the argument expression is dependent. (Also, no need to assert `ArgVal` three times.)

Reply:
I have added tests and i do not see any assert if any of [XYZ]DimVal is value-dependent.